### PR TITLE
upload marigold monocular depth estimation core files

### PR DESCRIPTION
**Pull request by Ranqing**
XR实验室上传Marigold单目深度估计模型

### CODE REVIEW

File: modelscope/models/cv/image_depth_estimation_marigold/networks/run.py [hunk 1]
```diff
@@ -0,0 +1,250 @@
+# Copyright 2023 Bingxin Ke, ETH Zurich. All rights reserved.
```
Comment (wenmengzhou) [inline modelscope/models/cv/image_depth_estimation_marigold/networks/run.py:1]: seems this file has not been used

File: modelscope/models/cv/image_depth_estimation_marigold/networks/marigold_model.py [hunk 2]
```diff
@@ -0,0 +1,382 @@
...
+        uncertainty (`None` or `np.ndarray`):
+            Uncalibrated uncertainty(MAD, median absolute deviation) coming from ensembling.
+    """
+
+    depth_np: np.ndarray
+    depth_colored: Image.Image
+    uncertainty: Union[None, np.ndarray]
+
+
+class MarigoldPipeline(DiffusionPipeline):
```
Comment (wenmengzhou) [inline modelscope/models/cv/image_depth_estimation_marigold/networks/marigold_model.py:61]: this pipeline should be moved to image_depth_estimation_pipeline

File: modelscope/pipelines/cv/image_depth_estimation_marigold_pipeline.py [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+from modelscope.utils.logger import get_logger
+
+logger = get_logger()
+
+
+@PIPELINES.register_module(
+    Tasks.image_depth_estimation, module_name=Pipelines.image_depth_estimation_marigold)
+class ImageDepthEstimationMarigoldPipeline(Pipeline):
+
+    def __init__(self, model: str, **kwargs):
```
Comment (wenmengzhou) [inline modelscope/pipelines/cv/image_depth_estimation_marigold_pipeline.py:20]: pls refer to https://github.com/modelscope/modelscope/blob/ade394d68c808b257028a961fc1bfa940c6bf7db/modelscope/pipelines/cv/card_detection_pipeline.py#L20 to add docstr

File: modelscope/models/cv/image_depth_estimation_marigold/networks/util/seed_all.py [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+# More information about the method can be found at https://marigoldmonodepth.github.io
+# --------------------------------------------------------------------------
+
+
+import numpy as np
+import random
+import torch
+
+
+def seed_all(seed: int = 0):
```
[the remark this answers is not in the document]
Comment (Ranqing) [reply modelscope/models/cv/image_depth_estimation_marigold/networks/util/seed_all.py:26]: the function has been replaced

File: modelscope/models/cv/image_depth_estimation_marigold/networks/run.py (hunk 1 above)
Comment (Ranqing) [reply modelscope/models/cv/image_depth_estimation_marigold/networks/run.py:1]: this file has been removed

File: modelscope/models/cv/image_depth_estimation_marigold/networks/marigold_model.py (hunk 2 above)
Comment (Ranqing) [reply modelscope/models/cv/image_depth_estimation_marigold/networks/marigold_model.py:61]: this is the Marigold model which is inherited from the DiffusionPipeline. Is it a solution that I change the name to MarigoldModel?

File: modelscope/models/cv/image_depth_estimation_marigold/depth_estimation_marigold_model.py [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+
+        if not torch.cuda.is_available():
+            raise Exception('GPU is required')
+        self.model_id = 'Damo_XR_Lab/cv_marigold_monocular-depth-estimation'
+        self.gpu_id = 0
+        self.device = torch.device('cuda:%d' % self.gpu_id)
+        self.checkpoint_path = os.path.join(model_dir, 'Marigold_v1_merged_2')
+        self.dtype = torch.float16
+
+        self.pipe = MarigoldPipeline.from_pretrained(self.checkpoint_path, torch_type=self.dtype)
```
[the remark this answers is not in the document]
Comment (Ranqing) [reply modelscope/models/cv/image_depth_estimation_marigold/depth_estimation_marigold_model.py:47]: Also, the pipe here is the Marigold model which is inherited from the DiffusionPipeline. If the solution above is ok, then self.pipe would be change to the model name.

File: modelscope/pipelines/cv/image_depth_estimation_marigold_pipeline.py (hunk 3 above)
Comment (Ranqing) [reply modelscope/pipelines/cv/image_depth_estimation_marigold_pipeline.py:20]: added

**Review by wenmengzhou — APPROVED** [review]

**Comment by wenmengzhou** [comment]
```shell
isort....................................................................Failed
- hook id: isort
- files were modified by this hook

Fixing modelscope/models/cv/image_depth_estimation_marigold/marigold.py
Fixing modelscope/models/cv/image_depth_estimation_marigold/marigold_utils.py
Fixing modelscope/pipelines/cv/image_depth_estimation_marigold_pipeline.py
Fixing tests/pipelines/test_image_depth_estimation_marigold.py

yapf.....................................................................Failed
- hook id: yapf
- files were modified by this hook
Trim Trailing Whitespace.................................................Failed
- hook id: trailing-whitespace
- exit code: 1
- files were modified by this hook

Fixing modelscope/pipelines/cv/image_depth_estimation_marigold_pipeline.py


```

linter test has benn failed, please refer to https://github.com/modelscope/modelscope/blob/master/docs/source/develop_cn.md#1-%E4%BB%A3%E7%A0%81%E9%A3%8E%E6%A0%BC to format code

**Comment by Ranqing** [comment]
> ```shell
> isort....................................................................Failed
> - hook id: isort
> - files were modified by this hook
> 
> Fixing modelscope/models/cv/image_depth_estimation_marigold/marigold.py
> Fixing modelscope/models/cv/image_depth_estimation_marigold/marigold_utils.py
> Fixing modelscope/pipelines/cv/image_depth_estimation_marigold_pipeline.py
> Fixing tests/pipelines/test_image_depth_estimation_marigold.py
> 
> yapf.....................................................................Failed
> - hook id: yapf
> - files were modified by this hook
> Trim Trailing Whitespace.................................................Failed
> - hook id: trailing-whitespace
> - exit code: 1
> - files were modified by this hook
> 
> Fixing modelscope/pipelines/cv/image_depth_estimation_marigold_pipeline.py
> ```
> 
> linter test has benn failed, please refer to https://github.com/modelscope/modelscope/blob/master/docs/source/develop_cn.md#1-%E4%BB%A3%E7%A0%81%E9%A3%8E%E6%A0%BC to format code

**Comment by Ranqing** [comment]
> ```shell
> isort....................................................................Failed
> - hook id: isort
> - files were modified by this hook
> 
> Fixing modelscope/models/cv/image_depth_estimation_marigold/marigold.py
> Fixing modelscope/models/cv/image_depth_estimation_marigold/marigold_utils.py
> Fixing modelscope/pipelines/cv/image_depth_estimation_marigold_pipeline.py
> Fixing tests/pipelines/test_image_depth_estimation_marigold.py
> 
> yapf.....................................................................Failed
> - hook id: yapf
> - files were modified by this hook
> Trim Trailing Whitespace.................................................Failed
> - hook id: trailing-whitespace
> - exit code: 1
> - files were modified by this hook
> 
> Fixing modelscope/pipelines/cv/image_depth_estimation_marigold_pipeline.py
> ```
> 
> linter test has benn failed, please refer to https://github.com/modelscope/modelscope/blob/master/docs/source/develop_cn.md#1-%E4%BB%A3%E7%A0%81%E9%A3%8E%E6%A0%BC to format code

linter

**Pull request merged**